lib/net.js

```diff
@@ -870,7 +870,8 @@ function afterWrite(status, handle, req, err) {
   if (self !== process.stderr && self !== process.stdout)
     debug('afterWrite', status);
 
-  self[kLastWriteQueueSize] = 0;
+  if (req.async)
+    self[kLastWriteQueueSize] = 0;
```

Comment:
What would happen if you just set this to the actual value of the write queue size at this point? Do you know that?

Reply:
This should always be 0 at this point. I could put an assert to confirm and run our test suite. The getter is obviously a bit more expensive.

Reply:
Hm … but what if multiple writes were scheduled and this callback ran after the first one finished? That’s a possibility, no?

Reply:
Only a single write can be scheduled at a time. `req.cb()` needs to be called (bottom of this function) before the next one will occur.